pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -70,6 +74,13 @@ pub mod pallet {
 	#[pallet::origin]
 	pub type Origin = GatewayOrigin;
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_idle(_now: BlockNumberFor<T>, max_weight: Weight) -> Weight {
+			Self::clear_invalid_session_ids(max_weight)
```

Comment:
@cdamian, I'm not sure if we need to remove the previous session IDs in `on_idle()`; I think we should just leave them there. We should remove them only when a message for the same hash has arrived and sessions are different. In that way, the fees/weight of doing that are computed at that moment.

As a rule of thumb, we should avoid `on_idle` unless it needs to be strictly needed.

Reply:
Not sure if that was a hard requirement when setting new routers. Maybe we can skip this and add some extra check when upserting inbound entries. I'll try this out on a different branch and get back to ya.

Reply:
Done separately [here](https://github.com/centrifuge/centrifuge-chain/pull/1965).